agent/doc/swdesign/README.md

```diff
@@ -1315,6 +1336,101 @@ Needs:
 - impl
 - utest
 
+#### WorkloadControlLoop handles restarts of workloads
+`swdd~workload-control-loop-handles-workload-restarts~1`
+
+Status: approved
+
+When the WorkloadControlLoop receives a new workload state, then the WorkloadControlLoop shall:
+- triggers the check of the permissibility of the restart
+- execute the restart of the workload it manages based on the result of the comparison
+
+Rationale:
+The execution of a restart of the workload depends on the workload state and the configured restart policy.
+
+Tags:
+- WorkloadControlLoop
+
+Needs:
+- impl
+- utest
+
+#### WorkloadControlLoop restarts workload with enabled restart policy
+`swdd~workload-control-loop-restarts-workload-with-enabled-restart-policy~1`
+
+Status: approved
+
+When:
+the workload's `ExecutionState` is `Succeeded(Ok)` or `ExecutionState` is `Failed(ExecFailed)` and the workload's configured `RestartPolicy` contains the value `ALWAYS` or
+the workload's `ExecutionState` is `Failed(ExecFailed)` and the workload's configured `RestartPolicy` contains the value `ON_FAILURE`,
+then the WorkloadControlLoop shall restart the workload.
+
+Rationale:
+The restart depends on the execution state of the workload.
+
+Tags:
+- WorkloadControlLoop
+
+Needs:
+- impl
+- utest
+- stest
+
+#### WorkloadControlLoop does not restart a workload with disabled restart policy
+`swdd~workload-control-loop-no-restart-with-disabled-restart-policy~1`
+
+Status: approved
+
+When the workload's configured `RestartPolicy` contains the value `NEVER`, then the WorkloadControlLoop shall not restart the workload.
+
+Rationale:
+The user has explicitly configured the workload to prevent its restart or to apply the default value of "NEVER" in the workload configuration.
+
+Tags:
+- WorkloadControlLoop
+
+Needs:
+- impl
+- utest
+- stest
```

Comment:
Do we really need this requirement as a separate one?
Maybe we can merge it with the one above by just stating in a comment that the "NEVER" policy is there for completeness and is handled as policy is never met.
(Normally one does not write negative requirements.)

Reply:
I can write it into a comment, but I thought it is more expressive to link it to the unit test explicitly as an extra requirement. I thought this is an exception here, because it is important that a workload is not restarted with policy `NEVER` and I did not want to cover this in a "weak" comment.

Reply:
I deleted this negative req and linked the one to the stest where inside the comment is now mentioned that with `NEVER` no restart happens.